src/account_manager.rs

```diff
@@ -20,46 +20,73 @@ use stronghold::Stronghold;
 ///
 /// Used to manage multiple accounts.
 #[derive(Default)]
-pub struct AccountManager {}
+pub struct AccountManager {
+    storage_path: PathBuf,
+}
 
 fn mutate_account_transaction<F: FnOnce(&Account, &mut Vec<Message>)>(
+    storage_path: &PathBuf,
     account_id: AccountIdentifier,
     handler: F,
 ) -> crate::Result<()> {
-    let mut account = crate::storage::get_account(account_id)?;
+    let mut account = crate::storage::get_account(&storage_path, account_id)?;
     let mut transactions: Vec<Message> = account.messages().to_vec();
     handler(&account, &mut transactions);
     account.set_messages(transactions);
-    let adapter = crate::storage::get_adapter()?;
-    adapter.set(account_id, serde_json::to_string(&account)?)?;
+    crate::storage::with_adapter(&storage_path, |storage| {
+        storage.set(account_id, serde_json::to_string(&account)?)
+    })?;
     Ok(())
 }
 
 impl AccountManager {
     /// Initialises a new instance of the account manager with the default storage adapter.
-    pub fn new() -> Self {
-        Default::default()
+    pub fn new() -> crate::Result<Self> {
+        Self::with_storage_path("./example-database")
+    }
+
+    /// Initialises a new instance of the account manager with the default storage adapter using the specified storage path.
+    pub fn with_storage_path(storage_path: impl AsRef<Path>) -> crate::Result<Self> {
+        Self::with_storage_adapter(
+            &storage_path,
+            crate::storage::get_adapter_from_path(&storage_path)?,
+        )
+    }
+
+    /// Initialises a new instance of the account manager with the specified adapter.
+    pub fn with_storage_adapter<S: StorageAdapter + Sync + Send + 'static>(
+        storage_path: impl AsRef<Path>,
+        adapter: S,
+    ) -> crate::Result<Self> {
+        crate::storage::set_adapter(&storage_path, adapter);
+        let instance = Self {
+            storage_path: storage_path.as_ref().to_path_buf(),
+        };
+        Ok(instance)
     }
 
     /// Sets the stronghold password.
     pub fn set_stronghold_password<P: AsRef<str>>(&self, password: P) -> crate::Result<()> {
-        let stronghold_path = crate::storage::get_stronghold_snapshot_path();
+        let stronghold_path = self
+            .storage_path
+            .join(crate::storage::stronghold_snapshot_filename());
         let stronghold = Stronghold::new(
             &stronghold_path,
             !stronghold_path.exists(),
             password.as_ref().to_string(),
             None,
         )?;
-        crate::init_stronghold(stronghold_path, stronghold);
+        crate::init_stronghold(self.storage_path.clone(), stronghold);
         Ok(())
     }
 
     /// Enables syncing through node events.
-    pub fn sync_through_events(&self) {
+    pub fn sync_through_events(&'static self) {
```

Comment:
I'm not sure about static lifetime here. Are you going to create an `AccountManager` with OnceCell or something similar?

Reply:
pushed a change to make it clone the storage_path instead of moving `self` to a closure, what do you think?